src/routes/manage_job/post.rs

```diff
@@ -0,0 +1,294 @@
+//! This file implements the `POST /bulk` endpoint.
+use crate::{
+	errors::ReacherError,
+	routes::check_email::{header::check_header, post::RetryOption},
+};
+use async_recursion::async_recursion;
+use check_if_email_exists::{
+	check_email as ciee_check_email, CheckEmailInput, CheckEmailInputProxy, CheckEmailOutput,
+	Reachable,
+};
+use sqlx::{Pool, Postgres};
+use warp::Filter;
+
+use std::{cmp::min, error::Error};
+
+use crate::routes::check_email::known_errors;
+use serde::{Deserialize, Serialize};
+use sqlxmq::{job, CurrentJob};
+
+const EMAIL_TASK_BATCH_SIZE: usize = 1;
+
+/// Errors that can happen during an email verification.
+#[derive(Debug)]
+enum CheckEmailError {
+	/// We get an `is_reachable` Unknown. We consider this internally as an
+	/// error case, so that we can do retry mechanisms (see select_ok & retry).
+	Unknown((CheckEmailOutput, RetryOption)),
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct TaskInput {
+	job_id: i32,
+	input: CheckEmailInput,
+}
+
+/// Endpoint request body.
+#[derive(Clone, Debug, Deserialize, Serialize, Default)]
+pub struct CreateBulkRequestBody {
+	input_type: String,
+	input: Vec<String>,
+	proxy: Option<CheckEmailInputProxy>,
+	hello_name: Option<String>,
+	from_email: Option<String>,
+	smtp_port: Option<u16>,
+}
+
+pub struct CreateBulkRequestBodyIterator {
+	body: CreateBulkRequestBody,
+	index: usize,
+	batch_size: usize,
+}
+
+impl IntoIterator for CreateBulkRequestBody {
+	type Item = CheckEmailInput;
+	type IntoIter = CreateBulkRequestBodyIterator;
+
+	fn into_iter(self) -> Self::IntoIter {
+		CreateBulkRequestBodyIterator {
+			body: self,
+			index: 0,
+			batch_size: EMAIL_TASK_BATCH_SIZE,
+		}
+	}
+}
+
+impl Iterator for CreateBulkRequestBodyIterator {
+	type Item = CheckEmailInput;
+
+	fn next(&mut self) -> Option<Self::Item> {
+		if self.index < self.body.input.len() {
+			let bounded_index = min(self.index + self.batch_size, self.body.input.len());
+			let to_emails = self.body.input[self.index..bounded_index].to_vec();
+			let mut item = CheckEmailInput::new(to_emails);
+
+			if let Some(name) = &self.body.hello_name {
+				item.set_hello_name(name.clone());
+			}
+
+			if let Some(email) = &self.body.from_email {
+				item.set_from_email(email.clone());
+			}
+
+			if let Some(port) = self.body.smtp_port {
+				item.set_smtp_port(port);
+			}
+
+			if let Some(proxy) = &self.body.proxy {
+				item.set_proxy(proxy.clone());
+			}
+
+			self.index = self.index + bounded_index;
+
+			Some(item)
+		} else {
+			None
+		}
+	}
+}
+
+impl Into<CheckEmailInput> for CreateBulkRequestBody {
```

Comment:
Still, this doesn't feel correct. `CreateBulkRequestBody` contains a `Vec<String>` of emails, while `CheckEmailInput` should only have 1 email. See comment above.

Reply:
Remove this.